tests/test_export.py

```diff
@@ -211,11 +212,14 @@ class View(ExportMixin, tables.SingleTableView):
         response = View.as_view()(build_request("/?_export=xlsx"))
         self.assertEqual(response.status_code, 200)
 
-        with NamedTemporaryFile(suffix=".xlsx") as tmp:
-            tmp.write(response.content)
-            tmp.seek(0)
-            wb = load_workbook(tmp.name)
-            self.assertIn(title, wb.sheetnames)
+        tmp = NamedTemporaryFile(suffix=".xlsx", delete=False)
+        tmp.write(response.content)
+        tmp.seek(0)
+        wb = load_workbook(tmp)
+        self.assertIn(title, wb.sheetnames)
+
+        tmp.close()
+        unlink(tmp.name)
```

Comment:
This won't be executed if anything between creating the file and this line fails.
Something like

```
tmp = NamedTemporaryFile...
try:
    tmp.write(...)
    ...
    self.assertIn(...)
finally:
    tmp.close()
    unlink(tmp.name)
```

would be a slightly more robust way to implement this.

Reply:
Yes, will use something like that.